internal/app/machined/pkg/runtime/v1alpha1/bootloader/dual/dual.go

```diff
@@ -0,0 +1,201 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package dual provides dual-boot bootloader implementation.
+package dual
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/siderolabs/go-blockdevice/v2/blkid"
+	"github.com/siderolabs/go-cmd/pkg/cmd"
+
+	"github.com/siderolabs/talos/internal/app/machined/pkg/runtime"
+	"github.com/siderolabs/talos/internal/app/machined/pkg/runtime/v1alpha1/bootloader/grub"
+	"github.com/siderolabs/talos/internal/app/machined/pkg/runtime/v1alpha1/bootloader/mount"
+	"github.com/siderolabs/talos/internal/app/machined/pkg/runtime/v1alpha1/bootloader/options"
+	"github.com/siderolabs/talos/internal/app/machined/pkg/runtime/v1alpha1/bootloader/sdboot"
+	"github.com/siderolabs/talos/internal/pkg/partition"
+	"github.com/siderolabs/talos/internal/pkg/uki"
+	"github.com/siderolabs/talos/pkg/imager/utils"
+	"github.com/siderolabs/talos/pkg/machinery/constants"
+)
+
+// Config describes a dual-boot bootloader.
+// this is a dummy implementation of the bootloader interface
+// allowing to install GRUB for BIOS and sd-boot for UEFI
+// so we only care about `RequiredPartitions()` and `Install()`.
+type Config struct{}
+
+// New creates a new bootloader.
+func New() *Config {
+	return &Config{}
+}
+
+// Install installs the bootloader.
+func (c *Config) Install(opts options.InstallOptions) (*options.InstallResult, error) {
+	if !opts.ImageMode {
+		return nil, fmt.Errorf("dual-boot bootloader is only supported in image mode")
+	}
+
+	var installResult *options.InstallResult
+
+	err := mount.PartitionOp(
+		opts.BootDisk,
+		[]mount.Spec{
```

Comment:
dumb question - can we use `Install` methods of both GRUB & SdBoot bootloaders here instead of duplicating the code?

Reply:
i could, but it does a little more things, so was trying to avoid it, also grub install is so many flags to pass/not pass based on UEFI